@here/harp-mapview/lib/geometry/TileGeometryLoader.ts

```diff
@@ -225,32 +242,46 @@ export class TileGeometryLoader {
      * Dispose of any resources.
      */
     dispose(): void {
-        this.reset();
+        this.clear();
         this.m_state = TileGeometryLoaderState.Disposed;
+        this.m_rejectFinishedPromise?.();
     }
 
     /**
      * Reset the loader to its initial state and cancels any asynchronous work.
      */
     reset(): void {
+        this.clear();
+
```

Comment:
Should the `this.m_rejectFinishedPromise`  be called first? Otherwise anyone waiting on the current promise will never be informed of the new promise and it will never be resolved / finished.

If you intend it, then at least document it.

Reply:
reset() does not change the state of the loader to any final state (cancelled, finished or disposed). It's done to reload the tile again (e.g. when tile is dirty), so waiters can still wait until the new tile geometry is loaded. I'll add a note to clarify it.

Reply:
I am referring to the condition below, where the ` this.m_rejectFinishedPromise` is set to a new function in the case where the state is Finished/Canceled/Disposed.

But you anyway call the promise in those cases, so I now see that it isn't an issue.

Reply:
Actually, I notice this method is never tested in the `TileGeometryLoaderTest.ts`, so please add a test for it.

Reply:
I've added tests to cover the reset method.